### PR TITLE
HRW: Adds an optional list of query params to remove/keep

### PR DESCRIPTION
**Pull request by zwoop**
This also adds the [I] / [INV] modifier to operators, where the logic should be INVerted. In this case, inverted means "remove all *but* the listed query parameters".

### CODE REVIEW

**Comment by zwoop** [comment]
I'm adding this to Project 10.0.x, but I think this would be fine to go into v10.1.0 as well.

File: doc/admin-guide/plugins/header_rewrite.en.rst [hunk 1]
```diff
@@ -1333,6 +1342,11 @@ The following ruleset removes any query parameters set by the client.::
    cond %{REMAP_PSEUDO_HOOK}
    rm-destination QUERY
 
+Remove only a few select query parameters:
+
+   cond %{REMAP_PSEUDO_HOOK}
+   rm-destination QUERY foo,bar
+
```
Comment (mlibbey) [inline doc/admin-guide/plugins/header_rewrite.en.rst:1349]: Perhaps also

```

Keep only a few select query parameters -- removing the rest:

   cond %{REMAP_PSEUDO_HOOK}
   rm-destination QUERY foo,bar [I]
```   
(if that is indeed how to do it)

File: doc/admin-guide/plugins/header_rewrite.en.rst [hunk 2]
```diff
@@ -775,6 +775,14 @@ changing the remapped destination, ``<part>`` should be used to indicate the
...
 parts for rm-destination are QUERY, PATH, and PORT.
 
+For the query parameter, this operator takes an optional second argument,
+which is a list of query parameters to remove (or keep with ``[INV]`` modifier).
+
+::
+
+  rm-destination QUERY <comma separate list of query parameter>
+
+
```
Comment (shukitchan) [inline doc/admin-guide/plugins/header_rewrite.en.rst:785]: Perhaps to make this a bit more comprehensive, should we make [INV] available to rm-header and rm-cookie as well?

Comment (shukitchan) [inline doc/admin-guide/plugins/header_rewrite.en.rst:785]: Perhaps to make this a bit more comprehensive, should we make [INV] available to rm-header and rm-cookie as well?

File: doc/admin-guide/plugins/header_rewrite.en.rst (hunk 1 above)
Comment (zwoop) [reply doc/admin-guide/plugins/header_rewrite.en.rst:1349]: Added, thanks!

File: doc/admin-guide/plugins/header_rewrite.en.rst (hunk 2 above)
Comment (zwoop) [reply doc/admin-guide/plugins/header_rewrite.en.rst:785]: Patches welcome? :-P

**Comment by zwoop** [comment]
[approve ci rocky]
[approve ci autest]

**Comment by zwoop** [comment]
[approve ci rocky]

**Comment by zwoop** [comment]
[approve ci]

**Comment by zwoop** [comment]
[approve ci autest]

**Comment by bneradt** [comment]
[approve ci autest]

**Comment by bneradt** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
@shukitchan I'm going to make a small change to the tokenizer, I think swoc::TextView has a more powerful mechanism here. I'll have it done tonight or tomorrow.

**Comment by shukitchan** [comment]
[approve ci autest]

**Review by shukitchan — APPROVED** [review]

**Comment by zwoop** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
[approve ci autest]

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**